lib/Test/MockFile.pm

```diff
@@ -186,11 +186,12 @@ sub _upgrade_barewords {
     my $handle;
     {
         no strict 'refs';
-        $handle = Symbol::qualify_to_ref( $args[1], caller(1) );
+        my $caller_pkg = caller(1);
+        $handle = *{"$caller_pkg\::$args[1]"};
     }
 
     # Check that the upgrading worked
-    ref $handle eq 'GLOB'
+    ref \$handle eq 'GLOB'
```

Comment:
wouldn't this be a ref to a ref?

Reply:
Here's the weird part:

* If you keep it `ref $handle eq 'GLOB'` with the bare-bones technique (which we revert to here), you will replicate the error (the test will fail)
* If you change it to `ref \$handle eq 'GLOB'` with the `Symbol.pm` technique, that also replicates the error (the test will fail)

In other words:

```
Symbol.pm + ref \$handle = crash
*{caller::...} + ref $handle = crash

Symbol.pm + ref $handle = no crash (except the test failure we're trying to fix now)
*{caller::...} + ref $handle = no crash (and maybe not the test failure we're trying to fix soon?)
```